AdyenCardScannerTests/CardScannerViewModelTests.swift

```diff
@@ -0,0 +1,207 @@
+//
+// Copyright (c) 2025 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import XCTest
+import AVFoundation
+@testable import AdyenCardScanner
+
+final class CardScannerViewModelTests: XCTestCase {
+
+    enum Constants {
+        static let mockCardImage = "adyen-card-iphone-capture"
+    }
+
+    var cardImageParser: CardImageParsingMock!
+    var captureSessionManager: CaptureSessionManagingMock!
+    var sut: CardScannerViewModel!
+
+    override func tearDownWithError() throws {
+        cardImageParser = nil
+        captureSessionManager = nil
+        sut = nil
+        try super.tearDownWithError()
+    }
+
+    func testVideoPreviewLayer() throws {
+        // Given
+        cardImageParser = CardImageParsingMock()
+        captureSessionManager = CaptureSessionManagingMock()
+
+        let expectedVideoPreviewLayer = AVCaptureVideoPreviewLayer()
+        captureSessionManager.videoPreviewLayer = expectedVideoPreviewLayer
+        sut = CardScannerViewModel(
+            cardImageParser: cardImageParser,
+            captureSessionManager: captureSessionManager
+        ) { _ in }
+
+        // When
+        let receivedVideoPreviewLayer = sut.videoPreviewLayer
+
+        // Then
+        XCTAssertTrue(expectedVideoPreviewLayer === receivedVideoPreviewLayer)
+    }
+
+    func testConfigureSessionShouldCallCaptureSessionManagerConfigureSession() throws {
+        // Given
+        cardImageParser = CardImageParsingMock()
+        captureSessionManager = CaptureSessionManagingMock()
+        sut = CardScannerViewModel(
+            cardImageParser: cardImageParser,
+            captureSessionManager: captureSessionManager
+        ) { _ in }
+
+        // When
+        sut.configureSession()
+
+        // Then
+        XCTAssertEqual(captureSessionManager.configureSessionCallsCount, 1)
+    }
+
+    func testStartCaptureSessionShouldCallCaptureSessionManagerStartCaptureSession() throws {
+        // Given
+        cardImageParser = CardImageParsingMock()
+        captureSessionManager = CaptureSessionManagingMock()
+        sut = CardScannerViewModel(
+            cardImageParser: cardImageParser,
+            captureSessionManager: captureSessionManager
+        ) { _ in }
+
+        // When
+        sut.startCaptureSession()
+
+        // Then
+        XCTAssertEqual(captureSessionManager.startCaptureSessionCallsCount, 1)
+    }
+
+    func testStartCaptureSessionShouldCallCaptureSessionManagerStopCaptureSession() throws {
```

Comment:
I am personally very biased against writing tests that capture internal implementation details instead of testing business logic / behaviors.

Approved. We can discuss the approach of what we test and why do we test it sometime in the future, this topic is quite broad.